kafka-plugins-0.10/src/main/java/io/cdap/plugin/sink/KafkaBatchSink.java

```diff
@@ -138,19 +142,23 @@ public void destroy() {
    * Kafka Producer Configuration.
    */
   public static class Config extends ReferencePluginConfig {
-
-    private static final String BROKERS = "brokers";
+    
     private static final String KEY = "key";
     private static final String TOPIC = KafkaBatchSink.TOPIC;
     private static final String FORMAT = "format";
     private static final String KAFKA_PROPERTIES = "kafkaProperties";
     private static final String COMPRESSION_TYPE = "compressionType";
 
-    @Name(BROKERS)
-    @Description("Specifies the connection string where Producer can find one or more brokers to " +
-      "determine the leader for each topic")
+    @Name(ConfigUtil.NAME_USE_CONNECTION)
+    @Nullable
+    @Description("Whether to use an existing connection.")
+    private Boolean useConnection;
+
+    @Name(ConfigUtil.NAME_CONNECTION)
     @Macro
-    private String brokers;
```

Comment:
This is backward incompatible change that causes the unit test to fail, this is strange as the source and sink have basically same config but with different names, `kafkaBrokers` vs `brokers`. I think what we can do is to create a new `KafkaSinkConnectorConfig` object, which contains both fields, both `@Nullable`, where the `getBrokers` will be based on the config that is not null. In the widget json, make `kafkaBrokers` hidden so user can only see one config.

Reply:
done